dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -0,0 +1,747 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <Columns/ColumnConst.h>
+#include <Common/Arena.h>
+#include <DataStreams/WindowBlockInputStream.h>
+#include <Interpreters/ExpressionActions.h>
+#include <Interpreters/convertFieldToType.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+extern const int NOT_IMPLEMENTED;
+} // namespace ErrorCodes
+
+WindowBlockInputStream::WindowBlockInputStream(const BlockInputStreamPtr & input, const WindowDescription & window_description_)
+    : window_description(window_description_)
+{
+    children.push_back(input);
+    output_header = input->getHeader();
+    for (const auto & add_column : window_description_.add_columns)
+    {
+        output_header.insert({add_column.type, add_column.name});
+    }
+
+    if (window_description.window_functions_descriptions.size() * window_description.aggregate_descriptions.size() != 0)
```

Comment:
why not remove the define of `window_description.aggregate_descriptions` and just add todo `support agg function`?